maven/spec/dependabot/maven/version_spec.rb

```diff
@@ -115,295 +145,350 @@
-
-          it { is_expected.to eq(0) }
-        end
-
-        context "when dealing with case insensitivity" do
-          let(:version) { described_class.new("1.0.FINAL") }
-          let(:other_version) { described_class.new("1") }
-
-          it { is_expected.to eq(0) }
-        end
-
-        context "when dealing with case insensitivity 2" do
-          let(:version) { described_class.new("1.something") }
-          let(:other_version) { described_class.new("1.SOMETHING") }
-
-          it { is_expected.to eq(0) }
-        end
-
-        context "when dealing with post releases" do
-          let(:version) { described_class.new("1-sp") }
-          let(:other_version) { described_class.new("1-ga") }
+    context "with equivalent shortened qualifiers" do
+      let(:versions) do
+        [
+          ["1alpha-0", "1a0", 0],
+          ["1beta-0", "1b0", 0],
+          ["1milestone-0", "1m0", 0]
+        ]
+      end
 
-          it { is_expected.to eq(1) }
+      it "returns 0" do
+        versions.each do |input|
+          version1, version2, result = input
+          version = described_class.new(version1)
+          other_version = described_class.new(version2)
+          expect(version <=> other_version).to eq(result)
+          expect(other_version <=> version).to eq(-result)
         end
+      end
+    end
 
-        context "when dealing with post releases 2" do
-          let(:version) { described_class.new("1-sp.1") }
-          let(:other_version) { described_class.new("1-ga.1") }
+    context "with dot, hyphen and digit / qualifier transitions as separators" do
+      let(:versions) do
+        [
+          ["1alpha.z", "1alpha-z", 0],
+          ["1alpha1", "1alpha-1", 0],
+          ["1alpha-1", "1alpha.1", -1],
+          ["1beta.z", "1beta-z", 0],
+          ["1beta1", "1beta-1", 0],
+          ["1beta-1", "1beta.1", -1],
+          ["1-a", "1a", 0],
+          ["1-a", "1.a", 0],
+          ["1-b", "1-b-1", -1],
+          ["1-b-1", "1-b.1", -1],
+          ["1sp.z", "1sp-z", 0],
+          ["1sp1", "1sp-1", 0],
+          ["1sp-1", "1sp.1", -1],
+          ["1rc.z", "1rc-z", 0],
+          ["1rc1", "1rc-1", 0],
+          ["1rc-1", "1rc.1", -1],
+          ["1milestone.z", "1milestone-z", 0],
+          ["1milestone1", "1milestone-1", 0],
+          ["1milestone-1", "1milestone.1", -1],
+          ["1snapshot.z", "1snapshot-z", 0],
+          ["1snapshot1", "1snapshot-1", 0],
+          ["1snapshot-1", "1snapshot.1", -1]
+        ]
+      end
 
-          it { is_expected.to eq(1) }
+      it "returns 0" do
+        versions.each do |input|
+          version1, version2, result = input
+          version = described_class.new(version1)
+          other_version = described_class.new(version2)
+          expect(version <=> other_version).to eq(result)
+          expect(other_version <=> version).to eq(-result)
         end
+      end
+    end
 
-        context "when dealing with null values (again)" do
```

Comment:
This test is incorrect. 1-sp-1 > 1-ga-1 as can be seen here: https://maven.apache.org/pom.html#version-order-specification . I've corrected this as part of the new implementation and this is now covered in the new tests.